src/glang/codegen/interfaces.py

```diff
@@ -64,10 +64,15 @@ class Kind(Enum):
         VALUE = 1
         MUTABLE_REF = 2
         CONST_REF = 3
+        # Mutable reference, but implicitly convertible to a const reference.
+        MUTABLE_OR_CONST_REF = 4
```

Comment:
This *sometimes valid* enum seems like it'll increase cyclomatic complexity... but this commit is very clean regardless, this seems like a good pragmatic approach.

Reply:
Well... it could always be valid, we just don't have explicit borrow syntax for it :)

Reply:
But yeah, I think this is easier to implement on top of what we have, compared to doing two lookups